The hardware queue scheduling is apparently configured with fixed
priorities, which creates a nasty fairness issue where traffic from one
CPU can starve traffic from all other CPUs.

Work around this issue by forcing all tx packets to go through one CPU,
until this issue is fixed properly.

Signed-off-by: Felix Fietkau <nbd@nbd.name>
---
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -4652,6 +4652,14 @@ static int mvneta_ethtool_set_eee(struct
 	return phylink_ethtool_set_eee(pp->phylink, eee);
 }
 
+static u16 mvneta_select_queue(struct net_device *dev, struct sk_buff *skb,
+			       struct net_device *sb_dev)
+{
+	/* XXX: hardware queue scheduling is broken,
+	 * use only one queue until it is fixed */
+	return 0;
+}
+
 static const struct net_device_ops mvneta_netdev_ops = {
 	.ndo_open            = mvneta_open,
 	.ndo_stop            = mvneta_stop,
@@ -4662,6 +4670,7 @@ static const struct net_device_ops mvnet
 	.ndo_fix_features    = mvneta_fix_features,
 	.ndo_get_stats64     = mvneta_get_stats64,
 	.ndo_do_ioctl        = mvneta_ioctl,
+	.ndo_select_queue    = mvneta_select_queue,
 	.ndo_bpf             = mvneta_xdp,
 	.ndo_xdp_xmit        = mvneta_xdp_xmit,
 };
